### ISSUE TITLE
Aligners with Auto Add Children and Nonpersistent items leave null entries

### ISSUE THREAD

**Issue by H3BO3** [Frooxius/NeosPublic#1033]
Reported by Ukilop - When an aligner with AutoAddChildren is saved with nonpersistent items in it, null entries are left in the aligner when it is brought out again. Related to #1032

**Comment by Frooxius** [comment]
This is not related to #1032 and it's not a bug either. The children are never removed, they're just not saved in the first place, so references to them are lost. 

I could add a cleanup function that runs on-load and removes any null entries, but it wouldn't be able to distinguish entries that are meant to be null as spacers and which are null because the target wasn't saved, so that could break some cases. Maybe as an optional toggle.

**Comment by H3BO3** [comment]
Perhaps. Isn't the whole point of AutoAddChildren to manage the children automatically? Why can't that be the toggle? If they don't exist then it should refresh the system and remove them if the flag is true.

**Comment by Frooxius** [comment]
No. The point of AutoAddChildren is to automatically add and remove children during the run of the world, but not much else beyond that. It doesn't interfere with the manually added or removed children, so you could have a setup that has a mix of both, in which case it could break by the change. Might be unlikely, but it's a consideration.

Perhaps having the behavior could be ok with just AutoAddChildren, but making sure we don't break any uses is important too.

**Comment by H3BO3** [comment]
Ah, I see. I suppose a toggle would be good then. I'll clarify something with Ukilop as well and get back to you.
